tortoisebeacon/handlers.go

```diff
@@ -344,27 +332,30 @@ func (tb *TortoiseBeacon) storeFirstVotes(message FirstVotingMessage, minerPK *s
 	tb.mu.Lock()
 	defer tb.mu.Unlock()
 
-	for _, vote := range message.ValidProposals {
-		if _, ok := tb.votesMargin[string(vote)]; !ok {
-			tb.votesMargin[string(vote)] = new(big.Int).Set(voteWeight)
+	for _, proposal := range message.ValidProposals {
+		p := string(proposal)
+		if _, ok := tb.votesMargin[p]; !ok {
+			tb.votesMargin[p] = new(big.Int).Set(voteWeight)
 		} else {
-			tb.votesMargin[string(vote)].Add(tb.votesMargin[string(vote)], voteWeight)
+			tb.votesMargin[p].Add(tb.votesMargin[p], voteWeight)
 		}
 	}
 
-	for _, vote := range message.PotentiallyValidProposals {
-		if _, ok := tb.votesMargin[string(vote)]; !ok {
-			tb.votesMargin[string(vote)] = new(big.Int).Neg(voteWeight)
+	for _, proposal := range message.PotentiallyValidProposals {
+		p := string(proposal)
+		if _, ok := tb.votesMargin[p]; !ok {
```

Comment:
if string([]byte) is passed as a key to the map (e.g. tb.votesMargin) it is optimized by compiler and memory is not copied. not a big deal probably but still cleaner

Reply:
ah. good to know. thanks!